sdks/swift/Mentat/Mentat/Mentat.swift

```diff
@@ -0,0 +1,188 @@
+        guard let success = result.ok else {
+            if let error = result.err {
+                let str = String(cString: error)
+                print("Error: \(str)")
+            }
+            return nil
+        }
+        return TypedValue(raw: success)
+    }
+
+    func set(date value: Date, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_timestamp_for_attribute_on_entid(self.intoRaw(), entid, attribute, value.toMicroseconds())
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func set(long value: Int64, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_long_for_attribute_on_entid(self.intoRaw(), entid, attribute, value)
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func set(reference value: Int64, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_entid_for_attribute_on_entid(self.intoRaw(), entid, attribute, value)
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func setKeywordReference(value: String, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_kw_ref_for_attribute_on_entid(self.intoRaw(), entid, attribute, value)
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func set(boolean value: Bool, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_boolean_for_attribute_on_entid(self.intoRaw(), entid, attribute, value ? 1 : 0)
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func set(double value: Double, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_double_for_attribute_on_entid(self.intoRaw(), entid, attribute, value)
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func set(string value: String, forAttribute attribute: String, onEntity entid: Int64) throws {
+        let result = store_set_string_for_attribute_on_entid(self.intoRaw(), entid, attribute, value)
+        guard let err = result.pointee.err else {
+            return
+        }
+
+        throw QueryError.executionFailed(message: String(cString: err))
+    }
+
+    func set(uuid value: UUID, forAttribute attribute: String, onEntity entid: Int64) throws {
```

Comment:
I'm not a big fan of these simple setter methods. I think they'll encourage code like:

```
store.set(double: 99.0, forAttribute: ":my/score", onEntity: 65536)?;
store.set(double: 128.5, forAttribute: ":my/height", onEntity: 65536)?;
```

which will run _two independent transactions_, dramatically increasing the amount of data written (one timestamp and tx metadata for each!) and thrashing the disk.

I'd even go so far as to say: expose `InProgress` and move these methods there, or delete them altogether.